axelor-business-project/src/main/java/com/axelor/apps/businessproject/web/ProjectTaskController.java

```diff
@@ -190,4 +191,16 @@ public void generatePurchaseOrder(ActionRequest request, ActionResponse response
       response.setCanClose(true);
     }
   }
+
+  public void updateChildrenProgress(ActionRequest request, ActionResponse response) {
+    try {
+      ProjectTask task = request.getContext().asType(ProjectTask.class);
+      BigDecimal progress = task.getProgress();
+      task = Beans.get(ProjectTaskRepository.class).find(task.getId());
+      Beans.get(ProjectTaskBusinessProjectService.class).updateChildrenProgress(task, progress);
+      response.setValue("projectTaskList", task.getProjectTaskList());
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
```

Comment:
I think we can move this code in the repository of ProjectTask instead of calling it in the onSave of the view and directly update the children task.

You can also create a service which will only be related to ProjectTask Progress Update.